File: app/src/main/java/linusfessler/alarmtiles/shared/views/TimeInputDialog.java
package linusfessler.alarmtiles.shared.views;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.DialogInterface;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;

import io.reactivex.disposables.CompositeDisposable;
import linusfessler.alarmtiles.R;

public class TimeInputDialog extends AlertDialog {

    private final TimeInput timeInput;
    private final CompositeDisposable disposable = new CompositeDisposable();

    @SuppressLint("InflateParams")
    public TimeInputDialog(@NonNull final Context context) {
        super(context);

        final View view = getLayoutInflater().inflate(R.layout.dialog_time_input, null);
        setView(view);
        timeInput = view.findViewById(R.id.time_input);
    }

    public TimeInputDialog(@NonNull final Context context, final InputMethodManager inputMethodManager) {
        this(context);
        setOnShowListener(dialog -> timeInput.showKeyboard(inputMethodManager));
    }

    public TimeInput getTimeInput() {
        return timeInput;
    }

    @Override
    protected void onStart() {
        super.onStart();
        disposable.add(timeInput.getTimeObservable()
                .subscribe(time -> {
                    final boolean enabled = time > 0;
                    getButton(DialogInterface.BUTTON_POSITIVE).setEnabled(enabled);
                }));
    }

    @Override
    protected void onStop() {
        super.onStop();
        disposable.clear();
    }
}

File: app/src/main/java/linusfessler/alarmtiles/shared/views/TimeInputDialog.kt
package linusfessler.alarmtiles.shared.views

import android.annotation.SuppressLint
import android.content.Context
import android.content.DialogInterface
import android.view.inputmethod.InputMethodManager
import androidx.appcompat.app.AlertDialog
import io.reactivex.disposables.CompositeDisposable
import linusfessler.alarmtiles.R

open class TimeInputDialog constructor(context: Context) : AlertDialog(context) {
    val timeInput: TimeInput

    private val disposable = CompositeDisposable()

    constructor(context: Context, inputMethodManager: InputMethodManager) : this(context) {
        setOnShowListener {
            timeInput.showKeyboard(inputMethodManager)
        }
    }

    init {
        @SuppressLint("InflateParams")
        val view = layoutInflater.inflate(R.layout.dialog_time_input, null)
        this.setView(view)
        timeInput = view.findViewById(R.id.time_input)
    }

    override fun onStart() {
        super.onStart()
        disposable.add(timeInput.timeObservable
                .subscribe {
                    getButton(DialogInterface.BUTTON_POSITIVE).isEnabled = it > 0
                })
    }

    override fun onStop() {
        super.onStop()
        disposable.clear()
    }
}